calicoctl/commands/node/run.go

```diff
@@ -181,15 +194,23 @@ Description:
 
 	// Create a mapping of environment variables to values.
 	envs := map[string]string{
-		"HOSTNAME": name,
```

Comment:
I don't believe we need to pass in HOSTNAME. 

The startup script honors both HOSTNAME and NODENAME and writes the name into a startup environments file that is used by rc.local  (and therefore by Felix and confd)